src/NATS.Client.Core/NatsTlsOpts.cs

```diff
@@ -23,5 +23,8 @@ public sealed record NatsTlsOpts
     /// <summary>When true, skip remote certificate verification and accept any server certificate</summary>
     public bool InsecureSkipVerify { get; init; }
 
+    /// <summary>When true, client initiates TLS connection immediately when connecting</summary>
+    public bool TlsFirst { get; init; }
```

Comment:
I think the technical term for this is "Implicit TLS", if we are looking to standardize around an industry-specific term.

In the future, if NATS introduces an Implict TLS port (like 4223 or whatever) this may define the behavior.  So we might want something like this:

```csharp
enum TlsMode {
  Auto,
  Disabled,
  Explicit,
  Implicit
}
```

That would allow us to drop the `Disabled` boolean and roll it into the enum as well

Reply:
Sounds like a sensible thing to me.

We should probably update the spec as well @Jarema?

Reply:
Added a comment to the spec - https://github.com/nats-io/nats-architecture-and-design/pull/247#discussion_r1360759283

Reply:
We may even want to expand out `Explicit` even further- within `Explicit` there is:

- `Prefer` - default when the `nats://` scheme is used - if the Server supports TLS, then use it, otherwise use plain-text
- `Require` - default when the `tls://` scheme is used- if the Server does not support TLS, then fail

Reply:
What's the difference between `prefered` and `auto`?

Reply:
`Auto` means use some criteria to decide.  The client logic as of today for `Auto` would be to use `Prefer` if a `nats://` scheme was used and `Require` if the `tls://` scheme was used.